server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -1100,12 +1122,35 @@ export function configureWebSocketServices(
 							: [contentBatch];
 
 						for (const content of contents) {
-							const signalMessage: ISignalMessage = {
-								clientId,
-								content,
-							};
+							let signalMessage: ISignalMessage;
+							let roomId: string;
+							const feature_submit_signals_v2 = "submit_signals_v2";
+
+							if (
+								supportedFeaturesMap.get(clientId)?.clientSupportedFeatures?.[
+									feature_submit_signals_v2
+								] &&
+								typeof content === "object" &&
+								"targetClientId" in content
```

Comment:
Consider writing this in a type guard function for clarity and stronger typing within (e.g. isTargetedSignal).  Maybe also verify typeof targetClientId if there's a concern it might be something weird.